debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/converters/sink/types/bson/DecimalType.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.converters.sink.types.bson;
+
+import java.math.BigDecimal;
+
+import org.apache.kafka.connect.data.Decimal;
+import org.apache.kafka.connect.errors.DataException;
+import org.bson.BsonDecimal128;
+import org.bson.BsonDouble;
+import org.bson.BsonValue;
+import org.bson.types.Decimal128;
+
+public class DecimalType extends AbstractBsonType {
+
+    public enum Format {
+        DECIMAL128, // for MongoDB v3.4+
+        LEGACYDOUBLE // results in double approximation
```

Comment:
I agree. Is there a way how Debezium can receive kegacy double on supported MongoDB versions?

Reply:
> I agree. Is there a way how Debezium can receive kegacy double on supported MongoDB versions?

In this context the enum represents the target type used for BigDecimal. SO I guess the short answer is "No"...  

Prior to 3.4 mongo didnt have a dedicated Decimal type, so you would only use doubles there. And doubles would be doubles for debezium.

Reply:
@jcechace which one to be removed precisely? `LEGACYDOUBLE`, `DECIMAL128` or both entirely?

Reply:
@jcechace  well LEGACYDOUBLE is to be removed as it's not really applicable (we don't support such ancient MongoDB version). Consequently there is no point in having just a single option, right? So that means DECIMAL128 will become the only (implicit) variant.

Reply:
Yes, IIUC the enum can be removed completely.

Reply:
done in an extra commit